test/urlMatcherFactorySpec.js

```diff
@@ -520,6 +520,21 @@ describe("urlMatcherFactory", function () {
       expect(m.format({ foo: 5, flag: true })).toBe("/5/1");
     });
 
+    it("should match types named only in params", function () {
+      var m = new UrlMatcher("/{foo}/{flag}", {
+        params: {
+          foo: {
+            type: 'int'
+          },
+          flag: {
+            type: 'bool'
+          }
+        }
+      });
+      expect(m.exec("/1138/1")).toEqual({ foo: 1138, flag: true });
+      expect(m.format({ foo: 5, flag: true })).toBe("/5/1");
+    });
+
```

Comment:
I got so caught up on the type tests in the urlMatcher that I forgot tests for url-less params. My bad!